test/syscall/syscall.py

```diff
@@ -628,7 +628,12 @@ def run(self):
         errfile = open(self.errfilename, 'w')
         if self.args.verbose:
             print('RUNNING: ' + str(callargs))
-        subret = subprocess.call(callargs, stdout=outfile, stderr=errfile)
+        # We have to overload the WIREDTIGER_CONFIG to ensure we perform
+        # fdatasync operations on checkpoints
+        my_env = os.environ.copy()
+        my_env["WIREDTIGER_CONFIG"] = "checkpoint_sync=true"
```

Comment:
Since `WIREDTIGER_CONFIG` overrides `wiredtiger_open` by design (so sysadmins can override a setting), it isn't sufficient to change this in the C code.

We could just change the Jenkins recipe, since it's Jenkins setting `WIREDTIGER_CONFIG` in the first place...